### PR TITLE
sample-sheet validation handled by Pipeline class.

### PR DESCRIPTION
**Pull request by charles-cowart**
Sample-sheet validation will now be handled by Pipeline class.
Pipeline class will use metapool to validate + perform any additional checks as needed by the Pipeline.

### CODE REVIEW

**Comment by antgonza** [comment]
This PR has conflicts, could you fix?

**Comment by antgonza** [comment]
Note that this one still has conflicts, can you fix?

File: sequence_processing_pipeline/Pipeline.py [hunk 1]
```diff
@@ -158,6 +160,49 @@ def add(self, job):
...
+        Performs additional validation for sample-sheet on top of metapool.
+        :param sample_sheet_path: Path to sample-sheet.
+        :return: If successful, an empty list of strings and a valid
+                 sample-sheet. If unsuccessful, a list of warning and error
+                 messages and None.
+        '''
+        # validate the sample-sheet using metapool package.
+        sheet = KLSampleSheet(sample_sheet_path)
+        msgs, val_sheet = quiet_validate_and_scrub_sample_sheet(sheet)
+        if val_sheet:
```
Comment (antgonza) [inline sequence_processing_pipeline/Pipeline.py:174]: What's the return value of `val_sheet` from `quiet_validate_and_scrub_sample_sheet`? If `None`, then (1) this should be `if val_sheet is None` and we don't need the two `return` statement; only one outside the if.

**Review by antgonza — COMMENTED** [review]
Looks good, one comment/question.

File: sequence_processing_pipeline/Pipeline.py (hunk 1 above)
Comment (charles-cowart) [reply sequence_processing_pipeline/Pipeline.py:174]: It's (None, sample-sheet-obj) if it's a valid sample-sheet and ([error messages], None) if not. Good point, thanks! I'll make the change.

Comment (charles-cowart) [reply sequence_processing_pipeline/Pipeline.py:174]: It's (None, sample-sheet-obj) if it's a valid sample-sheet and ([error messages], None) if not. Good point, thanks! I'll make the change.

**Pull request merged**